### ISSUE TITLE
Add a flow_sample function to sample optical flows and displacement fields

### ISSUE THREAD

**Issue by bnehoran** [pytorch/pytorch#36108]
_This issue is expanded from #24870 for reference and for additional discussions on implementation details. See that issue for context._

Add a function `flow_sample` to `torch.nn.functional` which mimicks `grid_sample`, but for sampling using only the residual displacement/flow. 
This eliminates the need to constantly add an identity grid to the flow/displacement field, which is imprecise, slow, and very [prone to user error](https://discuss.pytorch.org/t/warp-video-frame-from-optical-flow/6013).

The function signatures for both `flow_sample` and `grid_sample` will be nearly identical, and both will dispatch to the same underlying kernel (the existing `grid_sample` kernel).

This, along with #36107, would help simplify the implementation of convolutional networks to do Optical Flow Estimation and Image Registration tasks.

**Comment by hofingermarkus** [comment]
Having a dedicated `flow_sample` operation is not just a cosmetic change.
It can also have an impact on precision, however this only becomes relevant for large images and/or memory saving numerical data types.

For example:
Assume all pixels of a 4k Image move to by 1.05 pixel to the right.
While the optical flow ist 1.05 everywhere, the coordinates are [0...3829] -> 1.05, 2.05,...., 3831.05.
Formats like half precision have only 10-bits for the fraction, bfloat16 only 7-bits cannot store this relative accuray for larger pixels!
Hence, when using a coordinate format the accuracy of optical flow depends on the pixel positions, while it does not when specifying it directly as optical flow!
